### PR TITLE
Random seed split

### PR DESCRIPTION
**Pull request by joaorobson**
**Proposed changes**:
- Enable the user to set a seed to split the NLU data into test and training.
- Close https://github.com/RasaHQ/rasa/issues/4647

**Expected behavior**:
- When a user is splitting the NLU data, a new argument specifying a random seed (in sklearn, known  as random_state) could be passed, as shown below:
```rasa data split nlu --random-seed=42```

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by erohmensing** [comment]
@joaorobson is this still a draft PR or is it ready for review?

**Comment by joaorobson** [comment]
> @joaorobson is this still a draft PR or is it ready for review?

It is ready now. I just needed to update the docs and the changelog.

**Comment by erohmensing** [comment]
Ok, please mark it as ready for review, then.

File: rasa/nlu/training_data/training_data.py [hunk 1]
```diff
@@ -395,11 +395,15 @@ def build_nlg_stories_from_examples(examples) -> Dict[Text, list]:
                 ]
         return nlg_stories
 
-    def split_nlu_examples(self, train_frac) -> Tuple[list, list]:
+    def split_nlu_examples(self, train_frac, random_seed=None):
```
Comment (tabergma) [inline rasa/nlu/training_data/training_data.py]: Please, add types.
```suggestion
    def split_nlu_examples(self, train_frac: float, random_seed: Optional[int] = None) -> Tuple[list, list]:
```

File: rasa/nlu/training_data/training_data.py [hunk 2]
```diff
@@ -343,13 +343,13 @@ def validate(self) -> None:
                 )
 
     def train_test_split(
-        self, train_frac: float = 0.8
+        self, train_frac: float = 0.8, random_seed: int = None
```
Comment (tabergma) [inline rasa/nlu/training_data/training_data.py]:
```suggestion
        self, train_frac: float = 0.8, random_seed: Optional[int] = None
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -55,6 +55,8 @@ Improvements
 ------------
 - When NLU training data is dumped as Markdown file the intents are not longer ordered
   alphabetically, but in the original order of given training data
+- Added a random-seed parameter on the rasa data split nlu command allowing reproducible
+  nlu data splits.
```
Comment (tabergma) [inline CHANGELOG.rst]: We recently changed the way we add changelog entries. Please, have a look here: https://github.com/RasaHQ/rasa/tree/master/changelog

**Comment by joaorobson** [comment]
> Looks great  Just a few comments. Thanks for adding a test!
> 
> Please, update your branch and update the changelog entry (https://github.com/RasaHQ/rasa/tree/master/changelog).

Thanks. I guess that I've fixed everything that you requested.
